iceoryx_hoofs/source/posix_wrapper/named_pipe.cpp

```diff
@@ -136,8 +139,10 @@ NamedPipe& NamedPipe::operator=(NamedPipe&& rhs) noexcept
         IOX_DISCARD_RESULT(destroy());
         CreationPattern_t::operator=(std::move(rhs));
 
-        m_sharedMemory = std::move(rhs.m_sharedMemory);
-        m_data = std::move(rhs.m_data);
+        // for details see here iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/creation.hpp
+        m_sharedMemory = std::move(rhs.m_sharedMemory); // NOLINT(hicpp-invalid-access-moved)
+        // the rest are trivially-copyable, so no std::move
+        m_data = rhs.m_data;
```

Comment:
Please revert. This makes the code less robust. In case the pointer is changed to a smart pointer, e.g. a relative pointer, this assumption is not true anymore and makes refactoring harder.